core/src/main/java/org/apache/carbondata/scan/executor/impl/DetailQueryExecutor.java

```diff
@@ -36,8 +37,13 @@
   @Override public CarbonIterator<Object[]> execute(QueryModel queryModel)
       throws QueryExecutionException {
     List<BlockExecutionInfo> blockExecutionInfoList = getBlockExecutionInfos(queryModel);
-    return new DetailQueryResultIterator(blockExecutionInfoList, queryModel,
-        queryProperties.executorService);
+    if (queryModel.isVectorReader()) {
```

Comment:
can we create different implementation of AbstractQueryExecutor and new a different one in CarbonRecordReader?